app/controllers/AddressController.scala

```diff
@@ -331,80 +332,111 @@ class AddressController @Inject()(
             },
             addressSelectorDto => {
-              lookingUpAddress(typ, postcode, journeyData.addressLookupServiceDown) {
-                case AddressLookupSuccessResponse(recordSet) =>
-                  recordSet.addresses.find(_.id == addressSelectorDto.addressId.getOrElse("")) map {
-                    addressRecord =>
-                      val addressDto = AddressDto.fromAddressRecord(addressRecord)
-                      cacheSelectedAddressRecord(typ, addressRecord) flatMap {
+              journeyData.recordSet
+                .flatMap(_.addresses.find(_.id == addressSelectorDto.addressId.getOrElse(""))) match {
+                case Some(addressRecord) =>
+                  val addressDto = AddressDto.fromAddressRecord(addressRecord)
+                  cacheSelectedAddressRecord(typ, addressRecord) flatMap {
+                    _ =>
+                      cacheSubmittedAddressDto(typ, addressDto) map {
                         _ =>
-                          cacheSubmittedAddressDto(typ, addressDto) map {
-                            _ =>
-                              val postCodeHasChanged = !postcode
-                                .replace(" ", "")
-                                .equalsIgnoreCase(
-                                  personDetails.address.flatMap(_.postcode).getOrElse("").replace(" ", ""))
-                              (typ, postCodeHasChanged) match {
-                                case (PostalAddrType, true) => Redirect(routes.AddressController.enterStartDate(typ))
-                                case (PostalAddrType, false) =>
-                                  Redirect(routes.AddressController.showUpdateAddressForm(typ))
-                                case (_, true) => Redirect(routes.AddressController.enterStartDate(typ))
-                                case (_, false) =>
-                                  cacheSubmittedStartDate(typ, DateDto(LocalDate.now()))
-                                  Redirect(routes.AddressController.reviewChanges(typ))
-                              }
+                          val postCodeHasChanged = !postcode
+                            .replace(" ", "")
+                            .equalsIgnoreCase(personDetails.address.flatMap(_.postcode).getOrElse("").replace(" ", ""))
+                          (typ, postCodeHasChanged) match {
+                            case (PostalAddrType, true) => Redirect(routes.AddressController.enterStartDate(typ))
+                            case (PostalAddrType, false) =>
+                              Redirect(routes.AddressController.showUpdateAddressForm(typ))
+                            case (_, true) => Redirect(routes.AddressController.enterStartDate(typ))
+                            case (_, false) =>
+                              cacheSubmittedStartDate(typ, DateDto(LocalDate.now()))
+                              Redirect(routes.AddressController.reviewChanges(typ))
                           }
```

Comment:
Replace with a for comprehension as it is cleaner and has less indentation, like so|
```
for {
  _ <- cacheSelectedAddressRecord(typ, addressRecord)
  _ <- cacheSubmittedAddressDto(typ, addressDto)
} yield {
  val postCodeHasChanged = !postcode
    .replace(" ", "")
    .equalsIgnoreCase(personDetails.address.flatMap(_.postcode).getOrElse("").replace(" ", ""))
  (typ, postCodeHasChanged) match {
    case (PostalAddrType, true) => Redirect(routes.AddressController.enterStartDate(typ))
    case (PostalAddrType, false) =>
      Redirect(routes.AddressController.showUpdateAddressForm(typ))
    case (_, true) => Redirect(routes.AddressController.enterStartDate(typ))
    case (_, false) =>
      cacheSubmittedStartDate(typ, DateDto(LocalDate.now()))
      Redirect(routes.AddressController.reviewChanges(typ))
    }
}
```